### ISSUE TITLE
Copy-constructor

### ISSUE THREAD

**Issue by matpen** [rttrorg/rttr#14]
I am a new (since yesterday) user of the library, and first of all I wanted to tank you for such an useful tool.

I am starting now to explore the API and running some tests.
It seems that registration of a class requires a copy-constructor, even if `rttr::policy::ctor::as_raw_ptr` is used.

Small example attached. 
[test.zip](https://github.com/rttrorg/rttr/files/230224/test.zip)

This fails to build with a static assert.
Am I missing something?

Thank you in advance!

**Comment by matpen** [comment]
Furthermore, running the example from the [documentation](http://www.rttr.org/doc/master/structrttr_1_1policy_1_1ctor.html#a3a8150de98f16fdad329b85f43d32da1), it seems that the destructor is never called, and the variant stays valid even after calling destroy().

**Comment by acki-m** [comment]
Thanks for bringing up these two issues.
1)
This is an issue in the registration process of the constructor.
When you invoke it like this, then the default `policy::ctor::as_object` will be used.

```
rttr::registration::class_<MyStruct>("MyStruct")
            .constructor<>();
```
I have to think about this how to solve this correctly. So it is not your fault.

2)
Could catch. At the moment you have to retrieve the dtor of the raw type itself.
type::get<Foo>().destroy(var);
However, I can adjust this behaviour, so it will be done implicitly. Then this original code will work without leaking.

This two issues will be fixed in the upcoming release of 0.9.6

**Comment by matpen** [comment]
Thank you for your quick reply!
Will be waiting for 0.9.6 then. Keep up the good work!

**Comment by acki-m** [comment]
both issues are now fixed and merged to master in 0.9.6
